### DISCUSSION TITLE
cannot source my .env file

### DISCUSSION THREAD

**Discussion by arifulone** [Cyfrin/foundry-full-course-cu#109] [Q&A] [answered]
in lesson 6. i'm trying to source .env file to encrypt private_key & rpc_url . 
on the bash terminal i typed 
```source .env ```

but it through this error 
``` text 
PRIVATE_KEY: command not found
RPC_URL: command not found
```

**Comment by alymurtazamemon** [comment]
@arf92 Did you define the variables inside the env file like this `PRIVATE_KEY:<key>`?

It should be like `=` sign `PRIVATE_KEY=<key>`

**Reply by rohit-bhalerao-7** [reply, marked as answer]
Agreed, also @arf92 please don't add space on either side of equal to sign: variable' '=' 'value. It will also give you errors or won't work. It happened with me

**Reply by arifulone** [reply]
thank u guys . i actually did a tittle trick .  i put the entire command on a bash file and ran it . its lots easier than run a big command

**Reply by rohit-bhalerao-7** [reply]
Can you please explain it to me? I didn't get it.

**Reply by arifulone** [reply]
so i made a ```deploy.sh``` file  and put a long command to it . and save it 
like ```  
$ forge script script/DeploySimpleStorage.s.sol --rpc-url  https://eth-sepolia.g.alchemy.com/v2/xyzyourrpcurl --broadcast --private-key 123abcyourprivatekey``` .this one . 
instead of typing the command all the time , i simply open the ```deploy.sh``` file. on bash terminal .

**Reply by rohit-bhalerao-7** [reply]
Thanks, so you made the `deploy.sh` file like the `gitignore/env` file and added the code into it. And for opening it on bash terminal how do you do that?

**Reply by arifulone** [reply]
Just create a new file with dot sh extension. Like ```deploy.sh```.
and type your command in it and save,  no extra space,   quotation or line.  

Goto bash 
Type like this ```./deploy.sh``` 
fortunately,  this will not work.  
Bcz by default you cannot execute a  ```.sh``` file. 
You have change file permission. 

So,  to change it's permission or any file permission  
type on command line ```chmod +x deploy.sh```
now u can execute the file.  

Now,  on bash terminal  again type ```./deploy.sh```

. if any error comes fell free to share me

**Reply by rohit-bhalerao-7** [reply]
Thanks alot! Got it
